core/src/services/gcs/core.rs

```diff
@@ -136,6 +147,15 @@ impl GcsCore {
     }
 
     pub async fn sign_query<T>(&self, req: &mut Request<T>, duration: Duration) -> Result<()> {
+        if let Some(token) = &self.token {
```

Comment:
Do you think it's a good idea to return the token in `load_token`? We can set `expires` to `usize::MAX` for now. In the future, we might introduce `token_expires_in`.

Reply:
I don't quite understand what you mean here.

Is this in the sense of doing

```rust
pub async fn sign_query<T>(&self, req: &mut Request<T>, duration: Duration) -> Result<()> {
  if let Some(token) = &self.token {
    let loaded_token = self.load_token()?;
    // use the GoogleToken from `loaded_token` elsewhere
  }
  // ...
```

I'm a little confused because I thought the idea of having `token: Option<String>` meant we could bypass the use of `load_token` since it has already been explicitly provided, so there is nothing we need to "load" with the token loader etc.

Reply:
> Do you mean doing something like this?

Sorry for the confusion. I'm takling about:

```rust
async fn load_token(&self) -> Result<Option<GoogleToken>> {
    if let Some(token) = &self.token {
        return Ok(Some(GoogleToken::new(token, usize::MAX, &self.scope)));
    }

    let cred = { || self.token_loader.load() }
        .retry(&*BACKOFF)
        .await
        .map_err(new_request_credential_error)?;
    if let Some(cred) = cred {
        return Ok(Some(cred));
    }
    if self.allow_anonymous {
        return Ok(None);
    }
    Err(Error::new(
        ErrorKind::ConfigInvalid,
        "no valid credential found",
    ))
}
```

In this way, caller of `load_token` don't need to care about the token related logic. This will also make room for us to move the token set logic in `reqsign` directly.

Reply:
This makes sense, I had this locally, but I backtracked on it because I felt the inclusion of `scope` within the `GcsCore` was unnecessary because that is all part of `reqsign`.

I'll push what I had and let you take another look :+1: